packages/@ember/-internals/glimmer/lib/setup-registry.ts

```diff
@@ -17,7 +18,9 @@ export function setupApplicationRegistry(registry: Registry): void {
   // association won't leak
   registry.register('service:-dom-builder', {
     create(props) {
-      let env = getOwner(props).lookup('-environment:main') as { _renderMode: string };
+      let owner = getOwner(props);
+      assert('DomBuilderService is unexpectedly missing an owner', owner);
+      let env = owner.lookup('-environment:main') as { _renderMode: string };
```

Comment:
Do you mind adding a `// SAFETY` comment here explaining why we trust this to be safe?

Reply:
I'm actually not sure why offhand. I didn't write this, just got reformatted.

Reply:
Ahhh, I see. Fair!